gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/rest/HttpJsonServiceStubClassComposer.java

```diff
@@ -1103,6 +1143,43 @@ protected List<Expr> createOperationsStubInitExpr(
             .build());
   }
 
+  private Map<String, String> parseCustomHttpBindings(GapicContext context) {
+    Map<String, String> customHttpBindings = new HashMap<>();
+    com.google.api.Service service = context.serviceYamlProto();
+    if (service != null && service.getHttp() != null) {
+      for (HttpRule httpRule : service.getHttp().getRulesList()) {
+        customHttpBindings.put(httpRule.getSelector(), getValueBasedOnPatternCase(httpRule));
+      }
+    }
+    return customHttpBindings;
+  }
+
+  private Map<String, String> filterCustomHttpBindingsMap(
+      Map<String, String> customHttpBindings, Predicate<Map.Entry<String, String>> predicate) {
+    return customHttpBindings.entrySet().stream()
+        .filter(predicate)
+        .collect(Collectors.toMap(x -> x.getKey(), x -> x.getValue()));
+  }
+
+  private String getValueBasedOnPatternCase(HttpRule httpRule) {
+    switch (httpRule.getPatternCase().getNumber()) {
```

Comment:
It would be good if we can provide a reference to the http proto for these mappings.

Reply:
I'll remember to add it in later. I'm not sure if this is the best approach yet, so I haven't been adding any docs or tests yet.